content/courses/mobile/solana-mobile-dapps-with-expo.md

```diff
@@ -943,202 +1013,191 @@ through the code for each of them and then show you the entire file at the end:
-     description: string,
-     fileUri: string,
-   ) => {
-     if (!metaplex || !account || isLoading) return;
+Minting the NFT after the image and metadata have been uploaded is as simple as
+calling ``createNft` from `@metaplex-foundation/mpl-token-metadata`. Below shows
+the `createNFT` function tying everything together:
 
-     setIsLoading(true);
-     try {
-       const imageCID = await uploadImage(fileUri);
-       const metadataCID = await uploadMetadata(name, description, imageCID);
-
-       const nft = await metaplex.nfts().create({
-         uri: `https://ipfs.io/ipfs/${metadataCID}`,
-         name: name,
-         sellerFeeBasisPoints: 0,
-       });
-
-       setNftOfTheDay(nft.nft);
-     } catch (error) {
-       console.log(error);
-     } finally {
-       setIsLoading(false);
-     }
-   };
-   ```
+```tsx
+const createNFT = useCallback(
+  async (name: string, description: string, fileUri: string) => {
+    if (!umi || !account || isLoading) return;
+    setIsLoading(true);
+    try {
+      console.log(`Creating NFT...`);
+      const imageCID = await uploadImage(fileUri);
+      const metadataCID = await uploadMetadata(name, description, imageCID);
+      const mint = generateSigner(umi);
+      const transaction = createNft(umi, {
+        mint,
+        name,
+        uri: `https://${process.env.EXPO_PUBLIC_NFT_PINATA_GATEWAY_URL}/ipfs/${metadataCID}`,
+        sellerFeeBasisPoints: percentAmount(0),
+      });
+      await transaction.sendAndConfirm(umi);
+      const createdNft = await fetchDigitalAsset(umi, mint.publicKey);
+      setNftOfTheDay(createdNft);
+    } catch (error) {
+      console.error("Failed to create NFT:", error);
+    } finally {
+      setIsLoading(false);
+    }
+  },
+  [umi, account, isLoading, uploadImage, uploadMetadata],
+);
+```
 
 We'll put all of the above into the `NFTProvider.tsx` file. All together, this
 looks as follows:
 
 ```tsx
 import "react-native-url-polyfill/auto";
-import React, { ReactNode, createContext, useContext, useState } from "react";
+import React, {
+  ReactNode,
+  createContext,
+  useContext,
+  useState,
+  useCallback,
+  useMemo,
+} from "react";
+import { useConnection } from "./ConnectionProvider";
+import { Account, useAuthorization } from "./AuthProvider";
+import { useUmi } from "./MetaplexProvider";
 import {
-  Metaplex,
+  publicKey,
+  Umi,
   PublicKey,
-  Metadata,
-  Nft,
-  Sft,
-  SftWithToken,
-  NftWithToken,
-} from "@metaplex-foundation/js";
-import { useConnection } from "./ConnectionProvider";
-import { Connection, clusterApiUrl } from "@solana/web3.js";
+  generateSigner,
+  percentAmount,
+} from "@metaplex-foundation/umi";
+import { createUmi } from "@metaplex-foundation/umi-bundle-defaults";
+import { clusterApiUrl } from "@solana/web3.js";
 import { transact } from "@solana-mobile/mobile-wallet-adapter-protocol";
-import { Account, useAuthorization } from "./AuthProvider";
+import * as web3 from "@solana/web3.js";
```

Comment:
Just import each each item from web3 rather than * - it means people don't have to read `web3.` everywhere. Also it's what the other lessons do.